src/simulation/hdf_simulation_store.jl

```diff
@@ -282,27 +297,37 @@ function initialize_problem_storage!(
         @debug "Initialized optimizer_stats_datasets $problem ($num_columns, $num_stats)"
     end
 
-    emulator_group = _get_group_or_create(root, "emulation_model")
-    emulation_params = first(values(store.params.decision_models_params))
-    for type in STORE_CONTAINERS
-        group = _get_group_or_create(emulator_group, string(type))
-        for (key, reqs) in getfield(em_problem_reqs, type)
-            name = encode_key_as_string(key)
-            dataset = _create_dataset(group, name, reqs)
-            # Columns can't be stored in attributes because they might be larger than
-            # the max size of 64 KiB.
-            col = _make_column_name(name)
-            HDF5.write_dataset(group, col, string.(reqs["columns"]))
-            column_dataset = group[col]
-            datasets = getfield(get_em_data(store), type)
-            datasets[key] = HDF5Dataset(
-                dataset,
-                column_dataset,
-                get_resolution(emulation_params),
-                initial_time,
-            )
+    emulation_group = _get_group_or_create(root, "emulation_model")
+    for emulation_params in values(store.params.emulation_model_params)
+        for type in STORE_CONTAINERS
+            group = _get_group_or_create(emulation_group, string(type))
+            for (key, reqs) in getfield(em_problem_reqs, type)
+                name = encode_key_as_string(key)
+                dataset = _create_dataset(group, name, reqs)
+                # Columns can't be stored in attributes because they might be larger than
+                # the max size of 64 KiB.
+                col = _make_column_name(name)
+                HDF5.write_dataset(group, col, string.(reqs["columns"]))
+                column_dataset = group[col]
+                datasets = getfield(store.em_data, type)
+                datasets[key] = HDF5Dataset(
+                    dataset,
+                    column_dataset,
+                    # TODO: Josè, why is this broken?
+                    #get_resolution(emulation_params),
+                    get_resolution(first(values(params.decision_models_params))),
+                    initial_time,
+                )
+                container_key_lookup[encode_key_as_string(key)] = key
+            end
         end
     end
+
+    buf = IOBuffer()
```

Comment:
Let me know if what is happening here isn't obvious and I'll add an explanation.